### ISSUE TITLE
Warn about pseudo-docstrings

### ISSUE THREAD

**Issue by konstin** [astral-sh/ruff#7942]
The motivation is the following code from cpython:
```python
        if _sys.platform.startswith("aix"):
            """When the name contains ".a(" and ends with ")",
               e.g., "libFOO.a(libFOO.so)" - this is taken to be an
               archive(member) syntax for dlopen(), and the mode is adjusted.
               Otherwise, name is presented to dlopen() as a file argument.
            """
            if name and name.endswith(")") and ".a(" in name:
                mode |= _os.RTLD_MEMBER | _os.RTLD_NOW
```

What looks like a docstring is actually not a docstring, and we not lint or format it has a docstring. Instead we should show a warning that only module, functions and classes can have docstrings ([PEP 257](https://peps.python.org/pep-0257/#what-is-a-docstring)).
